/*
 Copyright 2012 Google Inc. All Rights Reserved.
 Author: blakely@google.com (Tim Blakely)

 Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at

     http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing, software
 distributed under the License is distributed on an "AS IS" BASIS,
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
*/

syntax = "proto2";

package bigbrain;

enum DiffEqSolver {
  NONE = 0;
  RUNGEKUTTA = 1;
  RUNGEKUTTA_LEGACY = 2;
}

message Position {
  optional float x = 1 [default = 0];
  optional float y = 2 [default = 0];
  optional float z = 3 [default = 0];
}

message Color {
  optional float r = 1 [default = .5];
  optional float g = 2 [default = .5];
  optional float b = 3 [default = .5];
}

// This is totally redundant, but protobufs don't have a typedef.  Can be
// renamed tuple, but that's not very descriptive
message Vector {
  optional float x = 1;
  optional float y = 2;
  optional float z = 3;
}

message Dimensions {
  optional float width = 1;
  optional float height = 2;
  optional float depth = 3 [default = 0];
}

message NetworkInfo {
  optional double time = 1;
  optional int64 next_uuid = 2;
  optional int32 random_seed = 3;
}
